salt/transport/tcp.py

```diff
@@ -215,45 +221,150 @@ class TCPPubClient(salt.transport.base.PublishClient):
+                if self.host and self.port:
+                    log.debug(
+                        "PubClient connecting to %r %r:%r", self, self.host, self.port
+                    )
+                    self._tcp_client = TCPClientKeepAlive(
+                        self.opts, resolver=self.resolver
+                    )
+                    stream = await asyncio.wait_for(
+                        self._tcp_client.connect(
+                            ip_bracket(self.host, strip=True),
+                            self.port,
+                            ssl_options=self.opts.get("ssl"),
+                            **kwargs,
+                        ),
+                        1,
+                    )
+                    self.unpacker = salt.utils.msgpack.Unpacker()
+                    log.debug(
+                        "PubClient conencted to %r %r:%r", self, self.host, self.port
+                    )
+                else:
+                    log.debug("PubClient connecting to %r %r", self, self.path)
+                    sock_type = socket.AF_UNIX
+                    stream = tornado.iostream.IOStream(
+                        socket.socket(sock_type, socket.SOCK_STREAM)
+                    )
+                    await asyncio.wait_for(stream.connect(self.path), 1)
+                    self.unpacker = salt.utils.msgpack.Unpacker()
+                    log.debug("PubClient conencted to %r %r", self, self.path)
+            except Exception as exc:  # pylint: disable=broad-except
+                log.warning(
+                    "TCP Publish Client encountered an exception while connecting to"
+                    " %s:%s %s: %r, will reconnect in %d seconds",
+                    self.host,
+                    self.port,
+                    self.path,
+                    exc,
+                    self.backoff,
+                )
+                if timeout and time.time() - start > timeout:
+                    break
+                await asyncio.sleep(self.backoff)
+            if timeout and time.time() - start > timeout:
+                break
+        return stream
+
+    async def _connect(self, timeout=None):
+        if self._stream is None:
+            self._closing = False
+            self._closed = False
+            self._stream = await self.getstream(timeout=timeout)
+            if self._stream:
+                # if not self._stream_return_running:
+                #    self.io_loop.spawn_callback(self._stream_return)
+                if self.connect_callback:
+                    self.connect_callback(True)
+            self.connected = True
+
+    async def connect(
+        self,
+        port=None,
+        connect_callback=None,
+        disconnect_callback=None,
+        timeout=None,
+    ):
+        if port is not None:
+            self.port = port
+        if connect_callback:
+            self.connect_callback = None
+        if disconnect_callback:
+            self.disconnect_callback = None
```

Comment:
```suggestion
        if connect_callback:
            self.connect_callback = None
        if disconnect_callback:
            self.disconnect_callback = None
```
Where are we setting `self.connect_callback` and `self.disconnect_callback` to the values passed in on the function?